crates/cli/commands/src/stage/drop.rs

```diff
@@ -113,6 +123,16 @@ impl<C: ChainSpecParser<ChainSpec: EthChainSpec + EthereumHardforks>> Command<C>
                 tx.clear::<tables::StorageChangeSets>()?;
                 tx.clear::<tables::Bytecodes>()?;
                 tx.clear::<tables::Receipts>()?;
+                if let Some(mut prune_checkpoint) =
+                    tx.get::<tables::PruneCheckpoints>(PruneSegment::Receipts)?
+                {
+                    prune_checkpoint.block_number = None;
+                    prune_checkpoint.tx_number = None;
+                    tx.put::<tables::PruneCheckpoints>(
+                        PruneSegment::Receipts,
+                        prune_checkpoint,
+                    )?;
+                }
```

Comment:
this code is repeated three times, it can be branched out into a function, that takes the tx and prune segment as param

Reply:
On it